web-scraper/src/crawler_setup.js

```diff
@@ -340,7 +341,7 @@ class CrawlerSetup {
      * @param {Object} environment
      * @returns {Function}
      */
-    async _handlePageFunction({ request, response, page, autoscaledPool }) {
+    async _handlePageFunction({ request, response, page, crawler }) {
```

Comment:
same here, would be nice to use the whole crawling context, but it will be difficult here and most likely impossible for some props

Reply:
Yeah, only DTOs or whatever you call "objects that only hold data" can be easily passed through the pipe to browser. Session is a pity, but I guess users can get `sessionId` from the `proxyInfo`. We could hack it by creating some representation of the Session class inside the browser, but I think that would be an overkill.

Reply:
If we care about the data only, we could also stringify with custom handler to allow dealing with cycles and re-parse that so we have POJOs without cycles. But for that I'd like to hear the list of things we care about, as the whole object is quite polluted with many things we probably don't care.

Reply:
Let's not overcomplicate it for now and pass only the stuff that's easy to do.